Source/SuperLinq/ToArray.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace SuperLinq;
+
+public static partial class SuperEnumerable
+{
+	/// <summary>
+	/// Creates an array from <see cref="IEnumerable{T}"/>.
```

Comment:
I think `.ToArray(int length)` is too ambiguous and deserves further discussion:
* Does the consumer expect an array of length `length` always? (my initial expectation)
* Does the consumer expect an array of the length of the sequence, and the `length` is an advisory for allocation performance? (your expectation)

I see both sides, and I'm not sure which would be the natural expectation. @Arithmomaniac do you have any thoughts on this?

Reply:
I think the first. (Compare to `new List<T>(3)` and `new T[3]`.

Reply:
Actually, in the `List` case it's an advisory, and in the `Array` case it's a definition - maybe we should match that semantics here as well.

Reply:
Do you suggest throwing an exception in ToArray if `length` doesn't match to data in `source`?
I think we should have `ToArray` and `ToList` consistent with each other.
I preferred a safe solution (like in List<>).
Maybe we should consider naming this parameter to make it more clear. I'm thinking about `initialCapacity` (similar to List ctor).

Reply:
Thanks- that explains my expectation. @konrad-gora please update to a fixed array size with no resizing. Throw an exception using `ThrowHelper` if the array is not large enough for `source`.

Reply:
If `source` is too short, then remaining items are left as `default`. If `source` is too long, then throw an exception.

Reply:
Don't forget to use `TryGetCollectionCount` and/or (in dotnet 6) `TryGetNonEnumeratedCount` when doing that array size check

Reply:
Sorry, but I couldn't agree. `ToList()` and `ToArray()` should be consistent with each other.

Reply:
I'd agree if they were consistent in dotnet. But in a library like this, you're trying to extend the language idiomatically. (Unlike, for example, https://github.com/louthy/language-ext).